### PR TITLE
#644: Expand ping cog to work on other websites

### PR DESCRIPTION
**Pull request by butteredwaffles**

### CODE REVIEW

File: ClemBot.Bot/bot/cogs/ping_pong_cog.py [hunk 1]
```diff
@@ -13,32 +14,60 @@ def __init__(self, bot: ClemBot):
...
 
-        embed.remove_field(1)
-        embed.insert_field_at(1, name="Discord HTTP", value=f"{discord_http_latency : 1.2f}ms")
+            embed.remove_field(1)
+            embed.insert_field_at(1, name="Discord HTTP", value=f"{discord_http_latency : 1.2f}ms")
 
-        await sent_message.edit(embed=embed)
+            await sent_message.edit(embed=embed)
+        else:
+            timeout = 3
```
Comment (butteredwaffles) [inline ClemBot.Bot/bot/cogs/ping_pong_cog.py:47]: Again, another subjective number on the timeout to enforce.

File: ClemBot.Bot/bot/cogs/ping_pong_cog.py [hunk 2]
```diff
@@ -13,32 +14,67 @@ def __init__(self, bot: ClemBot):
...
+            status = "success"
+            start = time.perf_counter()
+            try:
+                requests.get(url, timeout=timeout)
```
[the remark this answers is not in the document]
Comment (Jay-Madden) [reply ClemBot.Bot/bot/cogs/ping_pong_cog.py:51]: aiohttp must be used instead

File: ClemBot.Bot/bot/cogs/ping_pong_cog.py [hunk 3]
```diff
@@ -13,32 +14,67 @@ def __init__(self, bot: ClemBot):
...
+            ):
+                status = "invalid"
+            elapsed = time.perf_counter() - start
+
+            messages = {
+                "success": f"The request to '{url}' returned successfully in {elapsed * 1000 : 1.2f}ms.",
+                "conn_error": f"The server at '{url}' did not respond.",
+                "conn_timeout": f"The server at '{url}' did not respond within {timeout} seconds.",
+                "invalid": f"The URL '{url}' is invalid.",
+            }
```
Comment (Jay-Madden) [inline ClemBot.Bot/bot/cogs/ping_pong_cog.py:69]: We should never use strings as semantic/logical switches. stringly typed code is an anti pattern. Instead here we should just build the result string in each except handler. thats the easiest approach.

File: ClemBot.Bot/bot/cogs/ping_pong_cog.py [hunk 4]
```diff
@@ -13,32 +14,60 @@ def __init__(self, bot: ClemBot):
         self.bot = bot
 
     @ext.command(name="ping", aliases=["pong"])
+    @commands.cooldown(1, 10, commands.BucketType.user)
```
[the remark this answers is not in the document]
Comment (Jay-Madden) [reply ClemBot.Bot/bot/cogs/ping_pong_cog.py:17]: Good addition

**Review by Jay-Madden — APPROVED** [review]
some thoughts

**Review by Jay-Madden — DISMISSED** [review]
approved by accident lol

File: ClemBot.Bot/bot/cogs/ping_pong_cog.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (butteredwaffles) [reply ClemBot.Bot/bot/cogs/ping_pong_cog.py]: Sorry about that. Didn't realize

**Review by jkriste — APPROVED** [review]
LGTM! 🎉

**Pull request merged**